code/modules/food_and_drinks/drinks/drinks/drinkingglass.dm

```diff
@@ -16,517 +16,23 @@
 /obj/item/weapon/reagent_containers/food/drinks/drinkingglass/on_reagent_change()
 	cut_overlays()
 	if (reagents.reagent_list.len > 0)
```

Comment:
I don't like this because it takes out the color mixing and colors for specially named items that have no glass_icon_state [see grape soda]